File: py/915.py
class Solution:
    def partitionDisjoint(self, nums: List[int]) -> int:
        leftPos = 0
        leftMax = curMax = nums[0]
        for i in range(1, len(nums)):
            curMax = max(curMax, nums[i])
            if nums[i] < leftMax:
                leftMax, leftPos = curMax, i

        return leftPos + 1

File: rs/915.rs
impl Solution {
    pub fn partition_disjoint(nums: Vec<i32>) -> i32 {
        let (mut left_pos, mut left_max, mut cur_max) = (0, nums[0], nums[0]);
        for i in 1..nums.len() {
            cur_max = std::cmp::max(cur_max, nums[i]);
            if nums[i] < left_max {
                left_max = cur_max;
                left_pos = i;
            }
        }

        left_pos as i32 + 1
    }
}